Lib/test/support/__init__.py

```diff
@@ -2108,13 +2108,16 @@ def set_recursion_limit(limit):
     finally:
         sys.setrecursionlimit(original_limit)
 
-def infinite_recursion(max_depth=100):
+def infinite_recursion(max_depth=None):
     """Set a lower limit for tests that interact with infinite recursions
     (e.g test_ast.ASTHelpers_Test.test_recursion_direct) since on some
     debug windows builds, due to not enough functions being inlined the
     stack size might not handle the default recursion limit (1000). See
     bpo-11105 for details."""
-    if max_depth < 3:
+    if max_depth is None:
+        # Unoptimized number based on what works under a WASI debug build.
+        max_depth = 100 if python_is_optimized() else 50
```

Comment:
I don't think that `python_is_optimized()` is reliable. For example, I don't think that the function can detect that Python is built in debug mode on Windows with MSVC. I prefer to check also Py_DEBUG to be extra safe. I suggest:

```suggestion
        if not python_is_optimized() or Py_DEBUG:
            # Python built without compiler optimizations or in debug mode
            # usually consumes more stack memory per function call.
            # Unoptimized number based on what works under a WASI debug build.
            max_depth = 50
        else:
            max_depth = 100
```